### PR TITLE
optimize: limit redis mode all sessions

### LINKED ISSUE #4429: limit the number of queries in Redis storage mode
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
redis存储模式下，如果没有限制查询条数，tc的定时任务每次是捞出所有或某一批状态的事务，导致有可能造成oom的风险需要优化

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- Seata version:  
- OS :
- Others:

### PR DESCRIPTION
**Pull request by doubleDimple**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
limit the number of queries in Redis storage mode

### Ⅱ. Does this pull request fix one issue?

fixes #4429

### CODE REVIEW

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 1]
```diff
@@ -51,14 +53,17 @@
...
 import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_GLOBAL_XID;
+import static io.seata.core.constants.RedisKeyConstants.DEFAULT_LOG_QUERY_LIMIT;
+import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_BRANCH_STATUS;
+import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_GLOBAL_STATUS;
 import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_GLOBAL_GMT_MODIFIED;
 import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_BRANCH_GMT_MODIFIED;
-import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_GLOBAL_STATUS;
 import static io.seata.core.constants.RedisKeyConstants.REDIS_KEY_BRANCH_APPLICATION_DATA;
 
+
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:66]: 删了

Comment (doubleDimple) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:66]: get

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 2]
```diff
@@ -103,22 +117,23 @@ public static RedisTransactionStoreManager getInstance() {
...
-    /*
-    init map to constructor
+    /**
+     * init map to constructor
      */
     public RedisTransactionStoreManager() {
         super();
         initGlobalMap();
         initBranchMap();
+        logQueryLimit = CONFIG.getInt(STORE_REDIS_QUERY_LIMIT, DEFAULT_LOG_QUERY_LIMIT);
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:127]: 这个配置的key补充到application.example.yml和conifg.text中

Comment (doubleDimple) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:127]: get

**Review by a364176773 — APPROVED** [review]
LGTM

File: server/src/test/java/io/seata/server/session/redis/RedisTransactionStoreManagerTest.java [hunk 3]
```diff
@@ -45,6 +47,9 @@
  */
 @SpringBootTest
 public class RedisTransactionStoreManagerTest {
+
+    private Logger logger = LoggerFactory.getLogger(RedisTransactionStoreManagerTest.class);
```
Comment (caohdgege) [inline server/src/test/java/io/seata/server/session/redis/RedisTransactionStoreManagerTest.java:51]: private static final Logger LOGGER

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 4]
```diff
@@ -401,20 +416,16 @@ public GlobalSession readSession(String xid) {
...
-        return readSession(statuses, true);
-    }
-
     @Override
     public List<GlobalSession> readSession(GlobalStatus[] statuses, boolean withBranchSessions) {
         List<String> statusKeys = new ArrayList<>();
         for (int i = 0; i < statuses.length; i++) {
             statusKeys.add(buildGlobalStatus(statuses[i].getCode()));
         }
+        logQueryLimit =  resetLogQueryLimit(statusKeys);
```
Comment (caohdgege) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:425]: 这里不能直接给logQueryLimit赋值，不然会有并发访问的问题。应该是用个局部变量接收resetLogQueryLimit的返回值的

File: server/src/test/java/io/seata/server/session/redis/RedisTransactionStoreManagerTest.java (hunk 3 above)
Comment (doubleDimple) [reply server/src/test/java/io/seata/server/session/redis/RedisTransactionStoreManagerTest.java:51]: get

**Review by caohdgege — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
Does it conflict with the GlobalSessionParam pageNum parameter?

**Comment by slievrly** [comment]
> Does it conflict with the GlobalSessionParam pageNum parameter?

GlobalSessionParam calls readSession with the xid conditional argument, so there is no problem.

**Pull request merged**